### PR TITLE
AP_InertialSensor: simplify and correct AHRS_TRIM computation

### PR DESCRIPTION
**Pull request by jschall**

### CODE REVIEW

**Comment by jschall** [comment]
the trim calculations can be checked by running mavproxy and typing:
module load graph
graph degrees(ATTITUDE.roll) degrees(atan2(-RAW_IMU.yacc,-RAW_IMU.zacc))
graph degrees(ATTITUDE.pitch) degrees(atan2(RAW_IMU.xacc,sqrt(RAW_IMU.yacc**2+RAW_IMU.zacc**2)))

**Comment by jschall** [comment]
AHRS_TRIM_X and AHRS_TRIM_Y seem to both get set to .174533 sometimes with this patch.

**Comment by tridge** [comment]
applied, thanks Jon!

**Comment by brad112358** [comment]
Testing on real hardware with the current master branch, including this change leads me to conclude I was correct in my interpretation of the code.  We should be subtracting the offset, not adding it.  I see about 2 degrees of roll on the calibration surface with my setup with this PR and less than .2 degrees (noise)with the offset subtracted instead of added. @jschall can you weigh in on this please?  @tridge I assume you can just make a quick commit for this.  I'm happy to submit a pull request that changes one character if that's easier for anyone.  :-)

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp [hunk 1]
```diff
@@ -582,9 +582,14 @@ bool AP_InertialSensor::calibrate_accel(AP_InertialSensor_UserInteract* interact
...
         */
         Vector3f level_sample = samples[0][0];
+        level_sample.x *= new_scaling[0].x;
+        level_sample.y *= new_scaling[0].y;
+        level_sample.z *= new_scaling[0].z;
+        level_sample += new_offsets[0];
         level_sample.rotate(saved_orientation);
 
-        _calculate_trim(level_sample, trim_roll, trim_pitch);
+        trim_pitch = atan2(level_sample.x,pythagorous2(level_sample.y,level_sample.z));
```
Comment (magicrub) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp]: You guys are killing me! Please use atan2f() for floats. Now you made me re-scan the whole repo for this and I found a dozen more lingering around in various places. Please be careful.

**Comment by andyp1per** [comment]
This is closed, but is it actually fixed?

**Comment by brad112358** [comment]
Well, the original bug I reported (rotation performed in wrong order) is fixed, but the fix Jon made introduced a new bug (see my comments above) which causes very similar roll, pitch trim errors but now it affects most calibrations, not just orientation!=0.  My pr #2271 was correct, but not as concise as Jon's replacement here.  Unfortunately this one was merged before I found time to review it.

**Comment by andyp1per** [comment]
Can we reopen this if it's not fixed?

**Comment by rmackay9** [comment]
I'm not really involved in this issue but I saw a conversation between Jon & Tridge in which Jon said there was a bug but Tridge then fixed it.  I believe this patch is the fix:
https://github.com/diydrones/ardupilot/commit/b564ba0868292f7d0753ed3c645c95066c92a1e0

So I think it has been fixed but if someone finds it's not then we can revisit.

**Comment by andyp1per** [comment]
It looks like it might be:
-        level_sample += new_offsets[0];
+        level_sample -= new_offsets[0];

**Comment by brad112358** [comment]
Yes, it all looks correct now.  Neither Tridge nor Jon closed the loop here or with me when he made the second fix and I missed his commit where he fixed it even though I visually looked for it in the commits list.  Sorry if I added to the confusion when I said there was still a problem after the final fix was actually in.

**Pull request closed without merging**